homeassistant/components/drop/sensor.py

```diff
@@ -0,0 +1,501 @@
+        super().__init__("peak_flow_rate", device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the current flow rate."""
+        if self._device.peak_flow_rate is None:
+            return None
+        return round(self._device.peak_flow_rate, 1)
+
+
+class DROP_WaterUsedTodaySensor(DROP_Entity, SensorEntity):
+    """Monitors the total water used for the day."""
+
+    _attr_icon = WATER_ICON
+    _attr_native_unit_of_measurement = "gallons"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "water_used_today"
+
+    def __init__(self, device) -> None:
+        """Initialize the water used today sensor."""
+        super().__init__("water_used_today", device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the total used today."""
+        if self._device.water_used_today is None:
+            return None
+        return round(self._device.water_used_today, 1)
+
+
+class DROP_AverageWaterUsedSensor(DROP_Entity, SensorEntity):
+    """Monitors the average water used over the last 90 days."""
+
+    _attr_icon = WATER_ICON
+    _attr_native_unit_of_measurement = "gallons"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "average_water_used"
+
+    def __init__(self, device) -> None:
+        """Initialize the average water used sensor."""
+        super().__init__("average_water_used", device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the average used over the last 90 days."""
+        if self._device.average_water_used is None:
+            return None
+        return round(self._device.average_water_used, 1)
+
+
+class DROP_CapacityRemainingSensor(DROP_Entity, SensorEntity):
+    """Monitors the soft water capacity remaining on a softener."""
+
+    _attr_icon = WATER_ICON
+    _attr_native_unit_of_measurement = "gallons"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "capacity_remaining"
+
+    def __init__(self, device) -> None:
+        """Initialize the softener capacity remaining sensor."""
+        super().__init__("capacity_remaining", device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the softener capacity remaining ."""
+        if self._device.capacity_remaining is None:
+            return None
+        return round(self._device.capacity_remaining, 1)
+
+
+class DROP_CurrentSystemPressureSensor(DROP_Entity, SensorEntity):
+    """Monitors the current system pressure."""
+
+    _attr_icon = GAUGE_ICON
+    _attr_native_unit_of_measurement = "psi"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "current_system_pressure"
+
+    def __init__(self, device) -> None:
+        """Initialize the current system pressure sensor."""
+        super().__init__("current_system_pressure", device)
```

Comment:
There is a lot of duplication in sensor.py. Did you have a look at the EntityDescription class?

See https://developers.home-assistant.io/docs/core/entity/#entity-description

Example: https://github.com/home-assistant/core/blob/e594c19c1ecb9bc947b37d2af75e8d84f6a922e9/homeassistant/components/overkiz/sensor.py

Reply:
Yes, I looked at the class toward the end of initial development but I didn't see much of an advantage to refactoring it. I did clean up a lot of obvious duplication in 4123fbef4d086586be50e633ba84f619fcc3ff8e. I'm going to mark this resolved for the time being, and if you (or another reviewer) feels strongly that I need to use the EntityDescription class then we can reopen it and discuss further.